### PR TITLE
Ports Bay's stamina/sprinting intent system.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
Now this might actually be the last of the ports before I actually code things I swear.

Ports https://github.com/Baystation12/Baystation12/pull/25773 and adds sprinting by holding the shift key, also adding the option to add more movement intents for species with their own slowdown etc etc.

Todo:

- [x] Dionae, because of fucking course they get in the way

- [x] Test IPCs

- [x] Test simple mobs

- [x] Remove world logs

To-fix:

- [x] You apparently go mach 5 in space

- [x] Revert animal thing; just make a move decl with 0 delay for them

### CODE REVIEW

**Comment by OneOneThreeEight** [comment]
Any news on this?

**Comment by NonQueueingMatt** [comment]
will be done when simple mobs sre fixed

**Comment by NonQueueingMatt** [comment]
!review

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -123,7 +123,12 @@ default behaviour is:
 		now_pushing = FALSE
 		spawn(0)
 			. = ..()
-			if (!istype(AM, /atom/movable))
+			if (!istype(AM, /atom/movable) || AM.anchored)
+				if(confused && prob(50) && !MOVING_DELIBERATELY(src))
+					Weaken(2)
+					playsound(loc, "punch", 25, 1, -1)
+					visible_message("<span class='warning'>[src] [pick("ran", "slammed")] into \the [AM]!</span>")
```
Comment (Geevies) [inline code/modules/mob/living/living.dm:130]:
```suggestion
					visible_message(SPAN_WARNING("\The [src] [pick("ran", "slammed")] into \the [AM]!"))
```

File: code/datums/move_intent/move_intent.dm [hunk 2]
```diff
@@ -0,0 +1,42 @@
+// Quick and deliberate movements are not necessarily mutually exclusive
+#define MOVE_INTENT_DELIBERATE 1
+#define MOVE_INTENT_EXERTIVE   2
+#define MOVE_INTENT_QUICK      4
+
+/decl/move_intent
```
[the remark this answers is not in the document]
Comment (NonQueueingMatt) [reply code/datums/move_intent/move_intent.dm:6]: Bruh

**Comment by Alberyk** [comment]
Remove the overheating feature. It is really over the top.

**Pull request merged**